### PR TITLE
Upgrading Openshift Jenkins Plugin to use DSL version

### PR DESCRIPTION
**Pull request by myeung18**
Signed-off-by: Marco Yeung <myeung@redhat.com>

Old version will be no longer supported after OpenShift 3.11.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -160,6 +157,16 @@ You should see the application and it is started with web front-end like this:
...
+
+2) In Jenkins 2, running script in pipeline are subject to script security check (this pipeline was developed with Groovy script), and the script has to be approved before it is allowed to run.  When you start to run `3scale API publishing Pipeline`, you mostly will encounter "Scripts not permitted to use method ... " exceptions. If this happens, please login Jenkins as an administrator, go to "Manage Jenkins", "In-process Script Approval" page to approval the script.  You may need to do the approval multiple times as we found that Jenkins does the check on method level.  You only need to approve once on all those methods during the first run.  We suggest you to use persistent storage enabled Jenkins pod so that your approvals can be saved and reloaded if pod is restarted.
```
Comment (redhatHameed) [inline README.md:168]: can we add screen shot for better understanding here, that what methods and classes are required to approve. or we can put a separated readme file inside groovy scripts folder explaining more details. ?

File: Jenkinsfile [hunk 2]
```diff
@@ -337,28 +337,41 @@ def setEnvForDBModule(openShiftHost, openShiftToken, svcName, projName, mysqlUse
...
-                  srcStream: svcName,
-                  srcTag: sourceTag,
-                  destStream: svcName,
-                  destTag: destinationTag)
-
-    openshiftDeploy(namespace: projName,
-  			     deploymentConfig: svcName,
-			     waitTime: '300000')
-
+    script {
```
Comment (redhatHameed) [inline Jenkinsfile:360]: So everything required to put inside script {} tag this is only for using open shift plugin right ? is anything we required to change 3scale pipeline scipts ?

File: README.md (hunk 1 above)
Comment (myeung18) [reply README.md:168]: I would say screenshot is optional.  If we have other changes on the script in the future, the method names on the exception could be changed  as well. Then we need to keep updating the document.

File: Jenkinsfile (hunk 2 above)
Comment (myeung18) [reply Jenkinsfile:360]: script tag is needed to run/call plugin or Groovy like statement.

File: fisalert-service/src/main/resources/pipeline-app-build.yml [hunk 3]
```diff
@@ -11,7 +11,7 @@ parameters:
   value: https://github.com/RHsyseng/IntegrationApp-Automation.git
   required: true
 - name: GIT_BRANCH
-  value: master
+  value: master
```
Comment (bmozaffa) [inline fisalert-service/src/main/resources/pipeline-app-build.yml]: What's the change in this file? Just a space at the end?

File: fisuser-service/src/main/resources/pipeline-app-build.yml [hunk 4]
```diff
@@ -11,7 +11,7 @@ parameters:
   value: https://github.com/RHsyseng/IntegrationApp-Automation.git
   required: true
 - name: GIT_BRANCH
-  value: master
+  value: master
```
Comment (bmozaffa) [inline fisuser-service/src/main/resources/pipeline-app-build.yml]: What's the change in this file? Just a space at the end?

File: maingateway-service/src/main/resources/pipeline-app-build.yml [hunk 5]
```diff
@@ -11,7 +11,7 @@ parameters:
   value: https://github.com/RHsyseng/IntegrationApp-Automation.git
   required: true
 - name: GIT_BRANCH
-  value: master
+  value: master
```
Comment (bmozaffa) [inline maingateway-service/src/main/resources/pipeline-app-build.yml]: What's the change in this file? Just a space at the end?

**Review by bmozaffa — CHANGES_REQUESTED** [review]

File: fisalert-service/src/main/resources/pipeline-app-build.yml (hunk 3 above)
Comment (myeung18) [reply fisalert-service/src/main/resources/pipeline-app-build.yml]: not intended to change. 
It is just that when we do the testing, we need to use feature branch name, and before we make the PR, we need to change the name back to master. A space was there probably get removed by these steps.

Comment (myeung18) [reply fisalert-service/src/main/resources/pipeline-app-build.yml]: let me remove'em.

File: Jenkinsfile (hunk 2 above)
Comment (redhatHameed) [reply Jenkinsfile:360]: so it mean change required for 3scale scripts as well for DSL version. ? let me know if you will do that or I will make these changes in next PR.

File: fisalert-service/src/main/resources/pipeline-app-build.yml (hunk 3 above)
Comment (myeung18) [reply fisalert-service/src/main/resources/pipeline-app-build.yml]: done

File: Jenkinsfile (hunk 2 above)
Comment (myeung18) [reply Jenkinsfile:360]: don't get your question.
but if your pipeline runs with no error, no script tag is needed.

Comment (redhatHameed) [reply Jenkinsfile:360]: yes it runs without adding any scripts tag.  Hence resolving the conversation.

**Pull request merged**